rdm/0011-creators-contributors-metadata.md

```diff
@@ -0,0 +1,302 @@
+  }
+}
+```
+
+See #Alternatives below for the other take on this field.
+
+
+## How we teach this
+
+Once implemented and used, this schema will be self explanatory. The RFC will
+be a good location to understand the thinking that went into the decisions.
+
+
+## Drawbacks
+
+Less of a drawback, but more of an implication, is that various InvenioRDM
+modules that depend on a record schema will have to be amended to use this
+schema. If adopted by other services, this schema will require migrations.
+
+Although the generic approach is flexible, when it comes to defining a record
+**mapping** from this schema, there might be some more effort involved than
+in using a more set and explicit approach.
+
+
+## Alternatives
+
+The main alternative to the definition above is to split the `stakeholders`
+field into multiple fields like DataCite already does:
+
+```json
+"creators": {
+  "type": "array",
+  "description": "Creators in order of importance.",
+  "items": {
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "creator": { "$ref": "#/definitions/person_or_org" },
+    }
+  }
+},
+"contributors": {
+  "type": "array",
+  "description": "Contributors in order of importance.",
+  "items": {
+    "type": "object",
+    "additionalProperties": false,
+    "properties": {
+      "contributor": { "$ref": "#/definitions/person_or_org" },
+      "role": {
+        "type": "string",
+        "description": "Role the contributor has in this record.",
+        "enum": [
+          "ContactPerson",
+          "DataCollector",
+          "DataCurator",
+          "DataManager",
+          "Distributor",
+          "Editor",
+          "HostingInstitution",
+          "Other",
+          "Producer",
+          "ProjectLeader",
+          "ProjectManager",
+          "ProjectMember",
+          "RegistrationAgency",
+          "RegistrationAuthority",
+          "RelatedPerson",
+          "Researcher",
+          "ResearchGroup",
+          "RightsHolder",
+          "Sponsor",
+          "Supervisor",
+          "WorkPackageLeader",
+          "Other"
+        ]
+      }
+    },
+    "required": ["contributor", "role"]
+  }
+}
+```
+
+## Unresolved questions
+
+Is there a convention for field order? For properties fields, alphabetic order
+seems good. For other fields, having the `type` first helps parse the field I find.
+
+Using `ref`, should we wrap them or use them directly via `one_of`. For instance,
+should the pattern be:
+
+```json
+"<plural form>": {
+  "type": "array",
+  "description": "Many of this type.",
+  "items": {
+    "type": "object",
+    "properties": {
+      "<singular form>": { "$ref": "<definition>" }
+    }
+  }
+}
+```
+
+OR
+
+```json
+"<plural form>": {
+  "type": "array",
+  "description": "Many of this type.",
+  "items": {
+    "one_of": [{ "$ref": "<definition>" }]
+  }
+}
+```
+
+?
+
+Where should `owners` fit in this design (if at all)? The meaning of `owners`
```

Comment:
`owners` IMHO, is a "system" field (i.e. might contain a database user ID, and might be used for access control), in contrast to `creators` which is a bibliographic field.